### ISSUE TITLE
Incorrect "the following links could not be resolved" list

### ISSUE THREAD

**Issue by stanislav-tkach** [becheran/mlc#28]
**Describe the bug**
In some cases the resulting "the following links could not be resolved" list includes valid links. For example:
```
The following links could not be resolved:
./os_info/README.md (6, 87) => https://deps.rs/repo/github/stanislav-tkach/os_info.
./cli/README.md (6, 87) => https://deps.rs/repo/github/stanislav-tkach/os_info.
./README.md (6, 87) => https://deps.rs/repo/github/stanislav-tkach/os_info.
[ OK ] ./CHANGELOG.md (222, 10) => https://github.com/stanislav-tkach/os_info/compare/v0.7.0...v1.0. 

Result (90 links):

OK       87
Skipped  0
Warnings 0
Errors   3
```
There are only three failures in the log above, but the list includes four links.

**To Reproduce**
Unfortunately I'm unable to create a minimal reproducible example, moreover this issue isn't reproduced locally. However, it is stably reproduced on CI. Perhaps the MLC github action can affect this somehow?

Here are logs [with](https://github.com/stanislav-tkach/os_info/runs/1338649624) and [without](https://github.com/stanislav-tkach/os_info/runs/1338626756) debug output.

**Expected behavior**
I expect to see no "[ OK ]" links in the "the following links could not be resolved" section.

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04.1 LTS
 - Browser Firefox 82.0 (64-bit)
 - Version 0.13.11

**Comment by becheran** [comment]
Thanks for reporting. I will look into it. Do you use it in a public repo and can I see the failing/wrong checks?

**Comment by stanislav-tkach** [comment]
My mistake: I was sure, that I included the links to the logs! Sorry for that.

Here is the output from CI with the `--debug` flag: https://github.com/stanislav-tkach/os_info/runs/1338649624
And without: https://github.com/stanislav-tkach/os_info/runs/1338626756
The repository is public too: https://github.com/stanislav-tkach/os_info/pull/243

**Comment by stanislav-tkach** [comment]
By the way, https://deps.rs/repo/github/stanislav-tkach/os_info link is reported as "404 - Not Found", though the link is OK.

**Comment by becheran** [comment]
> By the way, https://deps.rs/repo/github/stanislav-tkach/os_info link is reported as "404 - Not Found", though the link is OK.

I could reproduce this "but" locally. The prblem is that the link returns 404 on a HEAD request: 
```
curl https://deps.rs/repo/github/stanislav-tkach/os_info --head
HTTP/2 404 
date: Wed, 11 Nov 2020 19:56:48 GMT
content-length: 0
set-cookie: __cfduid=d281a6e206188bd8b96d642babfad5d351605124607; expires=Fri, 11-Dec-20 19:56:47 GMT; path=/; domain=.deps.rs; HttpOnly; SameSite=Lax; Secure
cf-cache-status: DYNAMIC
cf-request-id: 065a7ae74500001181a6b41000000001
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report?s=QIYT4KZwhKEwiYRoDYKTddqGGSlYfKVvq9mpHbvbN40O6kbRt8EuawLsOKx2olFD5%2BgXPeNutuXTiqR%2FgEDW%2Bm8rzNapsECb"}],"group":"cf-nel","max_age":604800}
nel: {"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 5f0a941ed8771181-MIA
```
This is not an expected behavoir... though I might try a bit harder here and send another GET request which would give me a 200 then. I started with HEAD requests because they are so much cheaper... but yea, seems like there is a lot of unusual stuff out there...

> **Describe the bug**
> In some cases the resulting "the following links could not be resolved" list includes valid links. For example:
> 
> ```
> The following links could not be resolved:
> ./os_info/README.md (6, 87) => https://deps.rs/repo/github/stanislav-tkach/os_info.
> ./cli/README.md (6, 87) => https://deps.rs/repo/github/stanislav-tkach/os_info.
> ./README.md (6, 87) => https://deps.rs/repo/github/stanislav-tkach/os_info.
> [ OK ] ./CHANGELOG.md (222, 10) => https://github.com/stanislav-tkach/os_info/compare/v0.7.0...v1.0. 
> 
> Result (90 links):
> 
> OK       87
> Skipped  0
> Warnings 0
> Errors   3
> ```
> 
> There are only three failures in the log above, but the list includes four links.
> 
> **To Reproduce**
> Unfortunately I'm unable to create a minimal reproducible example, moreover this issue isn't reproduced locally. However, it is stably reproduced on CI. Perhaps the MLC github action can affect this somehow?
> 
> Here are logs [with](https://github.com/stanislav-tkach/os_info/runs/1338649624) and [without](https://github.com/stanislav-tkach/os_info/runs/1338626756) debug output.
> 
> **Expected behavior**
> I expect to see no "[ OK ]" links in the "the following links could not be resolved" section.
> 
> **Desktop (please complete the following information):**
> 
> * OS: Ubuntu 20.04.1 LTS
> * Browser Firefox 82.0 (64-bit)
> * Version 0.13.11

This one is harder. I could not reproduce it locally, yet. :( I will try to find a solution as soon as I know how to debug this....

**Comment by becheran** [comment]
> By the way, https://deps.rs/repo/github/stanislav-tkach/os_info link is reported as "404 - Not Found", though the link is OK.

This is now fixed with release v0.14.0.

**Comment by stanislav-tkach** [comment]
> This is now fixed with release v0.14.0.

Thanks!

**Comment by hoijui** [comment]
can be closed?
